_playwright-tests/UI/PopularRepo.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { closePopupsIfExist } from './helpers/helpers';
+import { deleteAllPopularRepos } from './helpers/deletePopularRepositories';
+
+test.describe('Popular Repositories', () => {
+  test('Add popular repos', async ({ page }) => {
+    // Ensure no popular repos are selected.
+    await deleteAllPopularRepos(page);
+
+    await navigateToRepositories(page);
+    await closePopupsIfExist(page);
+    await expect(page).toHaveTitle('Repositories - Content | RHEL');
+
+    await test.step('Select the Popular repos tab', async () => {
+      await page.getByRole('link', { name: 'Popular repositories' }).click();
+      await expect(page.getByTestId('popular_repos_table')).toBeVisible();
+      await page.getByRole('button', { name: 'Add selected repositories' });
+    });
+
+    await test.step('Select and add EPEL 9', async () => {
+      await page
+        .getByRole('row', { name: 'EPEL 9 Everything x86_64' })
+        .getByLabel('Select row 0', { exact: true })
+        .click();
+      await page.getByTestId('add-selected-dropdown-toggle-no-snap').click();
+      await page.getByRole('menuitem', { name: 'Add 1 repositories without snapshotting' });
+      await page
+        .getByRole('row', { name: 'EPEL 9 Everything x86_64' })
+        .getByTestId('remove_popular_repo')
+        .getByText('Remove');
```

Comment:
shouldn;t you expect it to be visible or something?

Or is this selector enough and it waits for it to be present?